app/commands/exercise/representation/search.rb

```diff
@@ -24,8 +24,10 @@ def call
 
   private
   def filter_with_feedback!
-    if with_feedback
+    if with_feedback && mentor
```

Comment:
Maybe be explicit and not rely on truthiness:

```suggestion
    if with_feedback && mentor.present?
```